### PR TITLE
feat: add filter bar to interests setting views

### PR DESCRIPTION
**Pull request by LeanderCretegny**
Add a filter to interests views in settings and onboarding to make searching for the wanted country, city or center of interest easier.

Testing: Run the code and try the filter by yourself and look for potential unexpected behavior. You can either go in the onboarding or in the settings to test the filter.

### CODE REVIEW

File: lib/widgets/wizard_widgets.dart [hunk 1]
```diff
@@ -1,8 +1,11 @@
+import 'dart:developer';
```
Comment (Thechi2000) [inline lib/widgets/wizard_widgets.dart:1]:
```suggestion
```

This import is unused.

File: lib/widgets/wizard_widgets.dart [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+
 import 'package:actualia/models/providers.dart';
 import 'package:actualia/viewmodels/providers.dart';
 import 'package:actualia/widgets/top_app_bar.dart';
 import 'package:actualia/utils/themes.dart';
 import 'package:fading_edge_scrollview/fading_edge_scrollview.dart';
+import 'package:flutter/cupertino.dart';
```
Comment (Thechi2000) [inline lib/widgets/wizard_widgets.dart:8]:
```suggestion
```

This import is unused.

**Review by Thechi2000 — COMMENTED** [review]
I'm not convinced of only matching prefixes. I think it would be more convenient to the user if the search can be matched anywhere in the string, especially with the incoming new predefined lists (https://github.com/ActualIA/app/pull/198).

**Comment by LeanderCretegny** [comment]
Regarding the prefix issue, you are not wrong, but there is no dart built-in function to do so (or maybe I have missed it, but I did not find one), which mean that we would need to code it ourselves and given the little amount of time we have left I would rather focus on other tasks.

**Comment by Thechi2000** [comment]
> Regarding the prefix match issue, you are not wrong, but there is no dart built-in function to do so (or maybe I have missed it, but I did not find one), which mean that we would need to code it ourselves and given the little amount of time we have left I would rather focus on other tasks. 

You can use the builtin `contains()` function, if I'm not mistaken.

> For the web part, we do not really care about it honestly.

I know, I just wanted to point it out in case it was a trivial fix.

**Comment by LeanderCretegny** [comment]
> You can use the builtin `contains()` function, if I'm not mistaken.
I did try, but it was not working and I incorrectly assumed the error was from the `contains()` function while it was in fact not. I have now made the change and use `contains()` instead of `matchAsPrefix()`

**Comment by Thechi2000** [comment]
I just tried, it seems to work just fine. However, you need to swap the two operands otherwise you would be trying to look for the item in the filter.

```dart
where((item) => item.$2.toLowerCase().contains(_filter.toLowerCase()))
```

**Review by Pacsort17 — APPROVED** [review]
Tested on Android, it works ! 
The design of the box could be a smoother, but nothing important.

File: lib/utils/themes.dart
Comment (LaClimSx) [inline lib/utils/themes.dart:1]: Not something you changed on this PR, but I notice on line 230 that body can never be null so this check doesn't do anything. Maybe you could remove it ?

**Review by LaClimSx — COMMENTED** [review]
Nothing to say on the code changed, good work! ,I've added a comment regarding existing code because it's fast to change even though you didn't technically change that part.

File: lib/widgets/wizard_widgets.dart
Comment (LaClimSx) [inline lib/widgets/wizard_widgets.dart:1]: Not something you changed on this PR, but I notice on line 230 that body can never be null so this check doesn't do anything. Maybe you could remove it ?

**Review by LaClimSx — APPROVED** [review]
Thanks for the change!

**Review by Thechi2000 — APPROVED** [review]

**Review by pmdlt — APPROVED** [review]
lgtm ! Run well on web + android via APK

**Comment by pmdlt** [comment]
@LeanderCretegny seems to be missing code coverage with the recents merges :/

**Comment by LeanderCretegny** [comment]
That is pretty annoying, I checked and the filter is almost completely covered by the tests. It means it is other parts of our code base that is lacking some coverage. I will open an issue that should be merged before this PR.

**Pull request merged**